src/App.test.jsx

```diff
@@ -0,0 +1,31 @@
+import { render, screen } from '@testing-library/react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { mockSingleCharacter } from './mocks/TestData';
+import { MemoryRouter } from 'react-router-dom';
+import App from './App';
+import userEvent from '@testing-library/user-event';
+
+const server = setupServer(
+  rest.get('https://api.disneyapi.dev/characters/:characterId', (req, res, ctx) => {
+    return res(ctx.json(mockSingleCharacter));
+  })
+);
+
+beforeAll(() => server.listen);
+afterAll(() => server.close);
+
+test.only('ensures character detail renders when a character is clicked', async () => {
+  render(
+    <MemoryRouter initialEntries={['/characters/5']} initialIndex={1}>
```

Comment:
Just to see if "olu mel" ended up on the page, you shouldn't need any initialEntries. And as App already has access to its routes, you probably don't need to wrap it in MemoryRouter.